wallet/createtx.go

```diff
@@ -943,22 +935,67 @@ func (w *Wallet) purchaseTickets(op errors.Op, req purchaseTicketRequest) ([]*ch
 		return nil, errors.E(op, errors.Invalid, "stakepool fee percent unset")
 	}
 
+	inSizes := make([]int, 0)
+	outSizes := make([]int, 0)
+	// TODO: use stake.SStxPKHMinOutSize after it has been updated.
+	commitmentScriptSize := 34
+	stakeSubmissionPkScriptSize := 0
+	addrID := [2]byte{req.ticketAddr.ScriptAddress()[0],
+		req.ticketAddr.ScriptAddress()[1]}
+	isP2SHAddr := chaincfg.IsScriptHashAddrID(addrID)
+	isP2PKHAddr := chaincfg.IsPubKeyHashAddrID(addrID)
+
+	if !isP2SHAddr && !isP2PKHAddr {
+		return nil, errors.E(op, errors.Invalid,
+			"ticket address must either be P2SH or P2PKH")
+	}
+	if isP2SHAddr {
+		stakeSubmissionPkScriptSize =
+			txsizes.P2SHPkScriptSize + 1 // + OP_SSTX
+	}
+	if isP2PKHAddr {
+		stakeSubmissionPkScriptSize =
+			txsizes.P2PKHPkScriptSize + 1 // + OP_SSTX
+	}
+
 	// Make sure that we have enough funds. Calculate different
 	// ticket required amounts depending on whether or not a
 	// pool output is needed. If the ticket fee increment is
 	// unset in the request, use the global ticket fee increment.
 	var neededPerTicket, ticketFee dcrutil.Amount
+	var estSize int
 	ticketFeeIncrement := req.ticketFee
 	if ticketFeeIncrement == 0 {
 		ticketFeeIncrement = w.TicketFeeIncrement()
 	}
 	if poolAddress == nil {
-		ticketFee = (ticketFeeIncrement * singleInputTicketSize) / 1000
-		neededPerTicket = ticketFee + ticketPrice
+		// A solo ticket has:
+		//   - a single input redeeming a P2PKH for the worst case size
+		//   - a P2PKH stake submission output
+		//   - a stake commitment output
+		//	 - a P2PKH change output
```

Comment:
The third output is not a P2PKH output.  It's an OP_SSTXCHANGE-tagged P2PKH or P2SH output.

There's also a tab character in this comment which doesn't match the rest.